react/src/components/CreateMapModal/CreateMapModal.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState } from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  FormGroup,
+  Label,
+  Input,
+} from 'reactstrap';
+import { Button } from '../../core-components';
+import styles from './CreateMapModal.module.css';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import useCreateProject from '../../hooks/projects/useCreateProject';
+import useAuthenticatedUser from '../../hooks/user/useAuthenticatedUser';
+import { useNavigate } from 'react-router-dom';
+
+type CreateMapModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+};
+
+// Yup validation schema
+const validationSchema = Yup.object({
+  name: Yup.string().required('Name is required'),
+  description: Yup.string().required('Description is required'),
+  system_file: Yup.string()
+    .matches(
+      /^[A-Za-z0-9-_]+$/,
+      'Only letters, numbers, hyphens, and underscores are allowed'
+    )
+    .required('Custom file name is required'),
+});
+
+const CreateMapModal = ({ isOpen, toggle }: CreateMapModalProps) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const { data: userData } = useAuthenticatedUser();
+  const { mutate: createProject, isLoading: isCreatingProject } =
+    useCreateProject();
+  const navigate = useNavigate();
+
+  const handleCreateProject = (projectData, actions) => {
+    createProject(projectData, {
+      onSuccess: (newProject) => {
+        navigate(`/project/${newProject.uuid}`);
+      },
+      onError: (err) => {
+        // Handle error messages while creating new project
+        setErrorMessage('An error occurred while creating the project.');
+        if (err?.response?.status === 409) {
+          setErrorMessage(
+            'That folder is already syncing with a different map.'
+          );
+        } else if (err?.response?.status === 500) {
+          setErrorMessage('Internal server error. Please contact support.');
+        }
+        actions.setSubmitting(false);
+      },
+    });
+  };
+
+  const handleSubmit = (values, actions) => {
```

Comment:
are the `actions` and `actions.setSubmitting` needed.

Reply:
They're not necessary, but keeping track of the submitting state might be helpful down the road when/if we add the feature to disable the Submit button, while the form is still handling the POST request? But I can take it out and we can add it back at that time?

Reply:
I might be missing something, but I think you have that case covered already by using  `isCreatingProject` and the `Button`'s isLoading parameter which adds a spinner and disables the button:
```
                <Button
                  size="short"
                  type="primary"
                  attr="submit"
                  isLoading={isCreatingProject}
                >
                  Create
                </Button>
```

Reply:
Oh yep you're right. Forgot we're using the core-style's Button wrapper now. I'll take the code out.